be/src/olap/rowset/segment_v2/column_reader.cpp

```diff
@@ -659,20 +662,33 @@ Status FileColumnIterator::_read_data_page(const OrdinalPageIndexIterator& iter)
     Slice page_body;
     PageFooterPB footer;
     _opts.type = DATA_PAGE;
-    RETURN_IF_ERROR(_reader->read_page(_opts, iter.page(), &handle, &page_body, &footer,
-                                       _compress_codec.get()));
-    // parse data page
-    RETURN_IF_ERROR(ParsedPage::create(std::move(handle), page_body, footer.data_page_footer(),
-                                       _reader->encoding_info(), iter.page(), iter.page_index(),
-                                       &_page));
+    auto reading_data_opts = _opts;
```

Comment:
You'd better add a comment here. Because you copied opts and opts has stats, if it is not a ptr, then something may wrong. But currently, it is right.